src/Shared/FileSystem/MSBuildOnWindowsFileSystem.cs

```diff
@@ -3,56 +3,78 @@
 
 using System.Collections.Generic;
 using System.IO;
+using System.Reflection;
+using Microsoft.Build.Framework;
 
 namespace Microsoft.Build.Shared.FileSystem
 {
     /// <summary>
-    /// Implementation of file system operations directly over the dot net managed layer
+    /// Implementation of file system operations on windows. Combination of native and managed implementations.
+    /// TODO Remove this class and replace with WindowsFileSystem. Test perf to ensure no regressions.
```

Comment:
Looks like you modified WindowsFileSystem. What more has to be done to replace this?

Reply:
Wait until nobody uses Windows anymore :)

Ideally this class should be implemented with Windows specific fast native APIs and used on Windows, instead of the presumably slower ManagedFileSystem. So somebody needs to make native implementations for the rest of the interface, test and prove better perf, and then delete `MSBuildOnWindowsFileSystem` in favor of just `WindowsFileSystem`.